core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceUnitTests.scala

```diff
@@ -0,0 +1,169 @@
+  }
+
+
+  "getWorkspaceById" should "return an exception without the workspace name when getWorkspace(WorkspaceName) is not found" in {
+    val workspaceFields: Future[Seq[(String, String)]] = Future.successful(List(("abc", "123")))
+    val datasource = mock[SlickDataSource]
+    when(datasource.inTransaction[Any](any(), any())).thenReturn(workspaceFields)
+    val service = spy(workspaceServiceConstructor(datasource)(defaultUserInfo))
+
+    doReturn(Future.failed(NoSuchWorkspaceException(WorkspaceName("abc", "123"))))
+      .when(service).getWorkspace(ArgumentMatchers.eq(WorkspaceName("abc", "123")), any(), any())
+
+    val workspaceId = "c1e14bc7-cc7f-4710-a383-74370be3cba1"
+    val exception = intercept[NoSuchWorkspaceException] {
+      Await.result(service.getWorkspaceById(workspaceId, WorkspaceFieldSpecs()), Duration.Inf)
+    }
+    assert(exception.workspace == workspaceId)
+    assert(!exception.getMessage.contains("abc"))
+    assert(!exception.getMessage.contains("123"))
+    verify(service).getWorkspace(ArgumentMatchers.eq(WorkspaceName("abc", "123")), any(), any())
+  }
+
+
+  "getWorkspaceById" should "return an exception without the workspace name when getWorkspace(WorkspaceName) fails access checks" in {
+    val workspaceFields: Future[Seq[(String, String)]] = Future.successful(List(("abc", "123")))
+    val datasource = mock[SlickDataSource]
+    when(datasource.inTransaction[Any](any(), any())).thenReturn(workspaceFields)
+
+    val service = spy(workspaceServiceConstructor(datasource)(defaultUserInfo))
+    doReturn(Future.failed(WorkspaceAccessDeniedException(WorkspaceName("abc", "123"))))
+      .when(service).getWorkspace(ArgumentMatchers.eq(WorkspaceName("abc", "123")), any(), any())
+
+    val workspaceId = "c1e14bc7-cc7f-4710-a383-74370be3cba1"
+    val exception = intercept[WorkspaceAccessDeniedException] {
+      Await.result(service.getWorkspaceById(workspaceId, WorkspaceFieldSpecs()), Duration.Inf)
+    }
+
+    assert(exception.workspace == workspaceId)
+    assert(!exception.getMessage.contains("abc"))
+    assert(!exception.getMessage.contains("123"))
+    verify(service).getWorkspace(ArgumentMatchers.eq(WorkspaceName("abc", "123")), any(), any())
+  }
+
+
+  "getWorkspaceById" should "return an exception with the workspace is when no workspace is found in the initial query" in {
+    val datasource = mock[SlickDataSource]
+    when(datasource.inTransaction[Any](any(), any())).thenReturn(Future.successful(List()))
+
+    val workspaceId = "c1e14bc7-cc7f-4710-a383-74370be3cba1"
+
+    val exception = intercept[NoSuchWorkspaceException] {
+      val service = workspaceServiceConstructor(datasource)(defaultUserInfo)
+      Await.result(service.getWorkspaceById(workspaceId, WorkspaceFieldSpecs()), Duration.Inf)
+    }
+
+    assert(exception.workspace == workspaceId)
+    assert(!exception.getMessage.contains("abc"))
```

Comment:
In this test case, "abc" and "123" are not passed at all, correct? If so, I don't think that lines 164 and 165 are testing anything.